x-pack/plugins/apm/public/components/fleet_integration/apm_agents/runtime_attachment/supported_agents/java_runtime_attachment.tsx

```diff
@@ -197,7 +199,9 @@ export function JavaRuntimeAttachment({ newPolicy, onChange }: Props) {
         apmVars?.java_attacher_discovery_rules?.value ?? '[]\n',
         [initialDiscoveryRule]
       )}
-      version={apmVars?.java_attacher_agent_version?.value || null}
+      version={
+        apmVars?.java_attacher_agent_version?.value || DEFAULT_AGENT_VERSION
```

Comment:
@kpollich can you confirm if the policy contains the `latest` value set by default after saving the integration? 

I think this will set `latest` in the text field value, but it never updates the policy version, am I right?

Reply:
Looks like the policy includes this block 

```
java_attacher:
        enabled: true
        discovery-rules:
          - include-vmarg: elastic.apm.attach=true
        download-agent-version: latest
```

Reply:
hmmmm I'm curious to understand when `onChange` is triggered to update this value in the policy, maybe the `EuiComboBox` call its `onChange`?

Reply:
When `onChange` is called as a result of enabling the Java Attacher field, that update operation now also includes the `latest` value for the `java_attacher_agent_version` field, e.g.

![image](https://user-images.githubusercontent.com/6766512/182187926-1ef9400a-b319-4099-9dce-ee180696b077.png)

Reply:
ahhhh got it. Amazing, thanks for that.